packages/block-library/src/post-date/edit.js

```diff
@@ -2,25 +2,93 @@
  * WordPress dependencies
  */
 import { useEntityProp, useEntityId } from '@wordpress/core-data';
+import { useState } from '@wordpress/element';
 import { __experimentalGetSettings, dateI18n } from '@wordpress/date';
+import { BlockControls, InspectorControls } from '@wordpress/block-editor';
+import {
+	ToolbarGroup,
+	ToolbarButton,
+	Popover,
+	DateTimePicker,
+	PanelBody,
+	CustomSelectControl,
+} from '@wordpress/components';
 import { __ } from '@wordpress/i18n';
 
-function PostDateDisplay() {
-	const [ date ] = useEntityProp( 'postType', 'post', 'date' );
+function PostDateEditor( { format, setAttributes } ) {
+	const [ siteFormat ] = useEntityProp( 'root', 'site', 'date_format' );
+	const [ date, setDate ] = useEntityProp( 'postType', 'post', 'date' );
+	const [ isPickerOpen, setIsPickerOpen ] = useState( false );
 	const settings = __experimentalGetSettings();
-
+	const is12Hour = /a(?!\\)/i.test(
+		settings.formats.time
+			.toLowerCase()
+			.replace( /\\\\/g, '' )
+			.split( '' )
+			.reverse()
+			.join( '' )
```

Comment:
Can we maybe add an example comment illustrating what `settings` looks like before and after transformation? The transformation is a bit abstract otherwise :sweat_smile:

Reply:
Yeah, done.